### PR TITLE
feat: make package/module callable

### LINKED ISSUE #321: Feat: make cli available as a python module
The CLI is available as "binary" scripts.

It could be made available as a python module, too.
To call via `python -m cyclonedx_py --help` and so on.

motivation:  The "binary" scripts are linked into the shell path of a user, and might be picked up by a shell. This process is unreliable and depends on outer factors(side effects). Other programs may override the environment, so that these scripts are no longer callable. The python env itself is a self containing environment, installed modules are always callable via `python -m`. Therefore, the module should make its CLI available as a callable module.

reminder: we are providing binary as `cyclonedx-py` alongside to `cyclonedx-bom`,
because other implementations like the one from `npm` also provicdea binary called `cyclonedx-bom`. and the systems will always overwrite the existing one with the one that was instaleld last. Therefore we need to provide alternatives like `cyclonedx-py`, `cyclonedx-node`, and so on.
SO this feature requests yet another alternative, one that is ecosystem(python)-specific and common  practice.

### PR DESCRIPTION
**Pull request by jkowalleck**
fixes #321

### CODE REVIEW

File: tests/integration/test_can_call_module.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    def test_callable_as_module(self) -> None:
+        args = [sys.executable, '-m', cyclonedx_py.__name__, '--help']
+        executable = sys.executable
+
+        # Test whether the call passed, is fair enough for now.
+        # Additional tests may come later, to check output etc.
+        returncode = subprocess.call(
+                executable=executable,
+                args=args,
+                shell=False,
```
[the remark this answers is not in the document]
Comment (jkowalleck) [reply tests/integration/test_can_call_module.py]: everything is fine.

File: tests/integration/test_can_call_module.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+
+import subprocess
```
[the remark this answers is not in the document]
Comment (jkowalleck) [reply tests/integration/test_can_call_module.py:20]: ignore

**Review by madpah — CHANGES_REQUESTED** [review]
Hey @jkowalleck - like the idea. Thanks for the PR.

Can you also add a documentation update to explain this new invocation method as per #321 ?

Also - would we consider this a breaking change given PEP-3102 syntax has been added?

**Comment by jkowalleck** [comment]
Sure, will add docs in the readme and other places.

Re: breaking change.
My intention was to add optional params to existing public api, so full backwards compatibility is given. Will check if I can improve the implementation.
Do you have any advice? Could you share your thoughts, @madpah ?

**Comment by jkowalleck** [comment]
actually, i do not have any idea where i could document the all via `python -m`. Please point me to it, @madpah .

regarding "is this a breaking change?"
A: nope, it still adds **optional** parameters to existing methods. No issue with backwards-compatibility, as the functionality is optional and usage is self-contained.

**Comment by madpah** [comment]
@jkowalleck - I've added Documentation to [CONTRIBUTING.md](https://github.com/CycloneDX/cyclonedx-python/blob/master/CONTRIBUTING.md) now - let me know if that is not enough to pop in some documentation changes.

Thanks on confirming NO breaking change 👍

**Comment by jkowalleck** [comment]
Thanks, @madpah .
I still do not know where to document alternative call methods. 
I see https://github.com/CycloneDX/cyclonedx-python/blob/master/docs/usage.rst talking about the params and such. and is points out to call `cyclonedx-bom`. But where is the note that `cyclonedx-py` for example is an alternative call method? I would add the `python -m cyclonedx-py` at the exact same spot. 

Could you point me to the spot i could update, @madpah ? I can't see the forest for the trees.

File: docs/usage.rst [hunk 3]
```diff
@@ -215,8 +223,8 @@ Requirements
 
 * :py:mod:`cyclonedx_py.parser.requirements.RequirementsParser`: Parses a multiline string that you provide that conforms
   to the ``requirements.txt`` :pep:`508` standard.
-* :py:mod:`cyclonedx_py.parser.requirements.RequirementsFileParser`: Parses a file that you provide the path to that conforms to the ``requirements.txt`` :pep:`508` standard. It supports nested
```
Comment (jkowalleck) [inline docs/usage.rst:218]: fixed a warning from the sphinx parser, here.

**Comment by jkowalleck** [comment]
added the missing docs. is this fine, @madpah  ?

**Comment by jkowalleck** [comment]
@madpah  please have a look at the proposed changes of this PR

**Review by madpah — APPROVED** [review]

**Pull request merged**